packages/ra-core/src/controller/useSortState.ts

```diff
@@ -16,12 +17,16 @@ const sortReducer = (state: Sort, field: string | Sort): Sort => {
         return field;
```

Comment:
I missed that part, but I think a reducer testing the type of the argument isn't very clean. You should enforce one call format. And since that hook is here to replace the Redux state for a ReferenceManyField, the call format should be identical to the one expected by the `setSort` action creator.